packages/teuchos/parameterlist/src/Teuchos_StandardParameterEntryValidators.cpp

```diff
@@ -294,6 +296,24 @@ int AnyNumberParameterEntryValidator::getInt(
   return 0; // Will never get here!
 }
 
+long long AnyNumberParameterEntryValidator::getLongLong(
+  const ParameterEntry &entry, const std::string &paramName,
+  const std::string &sublistName, const bool activeQuery
+  ) const
+{
+  const any &anyValue = entry.getAny(activeQuery);
+  if( acceptedTypes_.allowInt() && anyValue.type() == typeid(int) )
+    return as<long long>(any_cast<int>(anyValue));
+  if( acceptedTypes_.allowLongLong() && anyValue.type() == typeid(long long) )
```

Comment:
Does it matter in what order the types show up here?  It seems like you would want to check integer types in order of descending size, but the `anyValue.type()` check should at least ensure that you're not forcing `long long` to be `int`.

Reply:
I think in this block of code the order doesn't matter, because `anyValue.type()` will have only one exact value so only one of these can possibly evaluate to `true`.